### PR TITLE
Add ffmpeg 5.1.2

### PR DESCRIPTION
**Pull request by timmmyeo**
Add ffmpeg 5.1.2 dynamic executable with dependencies

### CODE REVIEW

**Review by StefanJum — CHANGES_REQUESTED** [review]
@timmmyeo this looks good, thank you.
Please remove the `usr/share` directory, since it is not needed, and add a `README.md` file where you give instructions on how to run the elf files both natively on Linux and on Unikraft (using the `run.sh` script).

File: ffmpeg/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+## Run ffmpeg natively on linux
+
+The binary is located in the `/usr/bin`` directory.
+Run it with the command:
+```console
```
Comment (StefanJum) [inline ffmpeg/README.md:5]:
```suggestion
Run it with the command:

```console
```

File: ffmpeg/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+## Run ffmpeg natively on linux
+
+The binary is located in the `/usr/bin`` directory.
```
Comment (StefanJum) [inline ffmpeg/README.md:3]:
```suggestion
The binary is located in the `./usr/bin` directory.
```

File: ffmpeg/README.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+```console
+./usr/bin/ffmpeg
+```
+
+## Run ffmpeg with Unikraft
+
+Make sure you have the *run-app-elfloader* repository [run-app-elfloader](https://github.com/unikraft/run-app-elfloader).
+
+Run the `run.sh` script from the [`run-app-elfloader` repository](https://github.com/unikraft/run-app-elfloader):
+```console
```
Comment (StefanJum) [inline ffmpeg/README.md:14]:
```suggestion
Run the `run.sh` script from the [`run-app-elfloader` repository](https://github.com/unikraft/run-app-elfloader):

```console
```

File: ffmpeg/README.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+./run.sh -r ../dynamic-apps/ffmpeg /usr/bin/ffmpeg
+```
+
+Run a simple encode job:
+```console
```
Comment (StefanJum) [inline ffmpeg/README.md:19]:
```suggestion
Run a simple encode job:

```console
```

**Review by StefanJum — CHANGES_REQUESTED** [review]
Few more comments.
@timmmyeo it seems like there are a lot more libraries added then there are dependencies. See which of them are not needed and remove them.

**Comment by timmmyeo** [comment]
> Few more comments. @timmmyeo it seems like there are a lot more libraries added then there are dependencies. See which of them are not needed and remove them.

I tried running the application without each one of the libraries, and it seems that ffmpeg actually needs them. All the files included seem to be needed.

File: ffmpeg/README.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+The binary is located in the `./usr/bin`` directory.
+Run it with the command:
+
+```console
+./usr/bin/ffmpeg
+```
+
+## Run ffmpeg with Unikraft
+
+Make sure you have the *run-app-elfloader* repository [run-app-elfloader](https://github.com/unikraft/run-app-elfloader).
```
Comment (StefanJum) [inline ffmpeg/README.md:12]:
```suggestion
Make sure you have the [`run-app-elfloader` repository](https://github.com/unikraft/run-app-elfloader) cloned.
```

File: ffmpeg/README.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
```
Comment (StefanJum) [inline ffmpeg/README.md:13]:
```suggestion
```

**Review by StefanJum — CHANGES_REQUESTED** [review]
Few more minor comments @timmmyeo, also please remove the parts of the commit description that are related to integrating review comments.
All good besides that.

**Review by StefanJum — APPROVED** [review]
Thank you @timmmyeo.
Reviewed-by: Stefan Jumarea <stefanjumarea02@gmail.com>
Approved-by: Stefan Jumarea <stefanjumarea02@gmail.com>

**Pull request merged**